### PR TITLE
Enhance README.md

### PR DESCRIPTION
**Pull request by igorbrasileiro**
#### What problem is this solving?
enhance readme.md to clarify the development of payment app #trivial 

#### How should this be manually tested?
There's no need to be tested

#### Checklist/Reminders

- [x] Updated `README.md`.
- [ ] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,64 +1,131 @@
...
+
+```javascript
+respondTransaction = (status) => {
+    $(window).trigger('transactionValidation.vtex', [status])
+}
 ```
+
+Notice that the `respondTransaction` function receives the `status` boolean value, which resolves (`status == true`) or rejects (`status == false`) an order placement.
+
+Hence, according to the `status` value, whenever the `transactionValidation.vtex` event occurs, the `respondTransaction` function will send the appropriate response to the checkout UI.
```
Comment (igorbrasileiro) [inline README.md:105]: Acho que faria mais sentido algo assim:

> `transactionValidation.vtex` event occurs, the Checkout UI will check the appropriate payment status.

File: README.md [hunk 2]
```diff
@@ -1,64 +1,131 @@
...
+- Handling the order payload.
+- Injecting external scripts.
+
+### Understanding the checkout response
+
+After the order placement, there's a validation step to check whether the transaction is approved or not.
+
+If the transaction is approved, the checkout UI redirects to the *Order Placed* page. Otherwise, it shows a warning pop-up.
+
+This validation is performed through the `transactionValidation.vtex` event of our *Payment* app, which sends a response to the checkout UI according to the following [method](https://github.com/vtex-apps/payment-authorization-app-example/blob/3e5742c87a2771998009cff4fecacb092bb3362b/react/index.js#L22):
```
Comment (igorbrasileiro) [inline README.md:95]: Acho que faria mais sentido algo assim:

> ... app, which sends the status of payment to the checkout UI ...

**Review by igorbrasileiro — COMMENTED** [review]
Deixei alguns comentários

File: README.md (hunk 2 above)
Comment (carolinamenezes) [inline README.md:95]:
```suggestion
In our *Payment* app, this validation is performed through the `transactionValidation.vtex` event, which, when triggered, sends the payment status to the checkout UI according to the following [method](https://github.com/vtex-apps/payment-authorization-app-example/blob/3e5742c87a2771998009cff4fecacb092bb3362b/react/index.js#L22):
```

File: README.md (hunk 1 above)
Comment (carolinamenezes) [inline README.md:105]:
```suggestion
Hence, whenever the `transactionValidation.vtex` event occurs, the Checkout UI checks the payment status and is redirected according to the `status` value provided to the `respondTransaction` function
```

File: README.md [hunk 3]
```diff
@@ -1,64 +1,131 @@
...
 const { appPayload } = this.props // This appPayload is a serialized JSON (as string).
 ```
 
-## Injecting external script
-
-In order to be able to run external scripts on your transaction app, you need to inject that script on the head of the checkout html. To do so, you have to do a DOM injection, for that you should do:
+The expected response is a JSON object as in the following:
 
+```json
 ```
```
Comment (carolinamenezes) [inline README.md:120]: @igorbrasileiro tem um exemplo para a gente colocar aqui? 🙈

File: README.md [hunk 4]
```diff
@@ -1,64 +1,135 @@
...
+Thus, to access this information, just run the following command:
+
+```javascript
 const { appPayload } = this.props // This appPayload is a serialized JSON (as string).
 ```
 
-## Injecting external script
-
-In order to be able to run external scripts on your transaction app, you need to inject that script on the head of the checkout html. To do so, you have to do a DOM injection, for that you should do:
+The expected response is a JSON object as in the following:
```
Comment (carolinamenezes) [inline README.md:117]:
```suggestion
The expected response is a JSON object consisting of the data needed for the Payment app to allow or deny an order placement. Thus, depending on the payment processes implemented in your store, the `appPayload` fields may vary.

Take the following example:
```

Comment (carolinamenezes) [reply README.md:117]: @igorbrasileiro faz sentido? KK

Comment (igorbrasileiro) [reply README.md:117]: Faz

File: README.md [hunk 5]
```diff
@@ -1,64 +1,135 @@
...
 
+```json
+{
+    "timeToCancel": 300,
+    "transactionApproveLinl": "https://..."
```
Comment (igorbrasileiro) [inline README.md:122]:
```suggestion
    "transactionApproveLink": "https://..."
```

**Review by paul0vinicius — APPROVED** [review]
LGTM

**Pull request merged**